### DISCUSSION TITLE
CodeQL Information

### DISCUSSION THREAD

**Discussion by bhuvi11** [community/community#24074] [Apps, API and Webhooks]
<p>Hello Team,</p>
<p>I am trying to use Codeql analysis and need information on few things</p>
<ol>
<li>Can I use codeql in bitbucket or any other git repository hosting service?</li>
<li>Difference between codeql runner and codeql CLI?</li>
</ol>

**Comment by jhutchings1** [comment, marked as answer]
<p>Hello!</p>
<blockquote>
<ol>
<li>Can I use codeql in bitbucket or any other git repository hosting service?</li>
</ol>
</blockquote>
<p>Please review the CodeQL license for the specific places you can run CodeQL for free. <a href="https://securitylab.github.com/tools/codeql/license" rel="noopener nofollow ugc">CodeQL License - OSS version - GitHub Security Lab</a> If you’re looking to analyze your private repository code, then please contact the GitHub sales team for more information. <a href="https://enterprise.github.com/contact" class="inline-onebox" rel="noopener nofollow ugc">Contact us - GitHub Enterprise</a></p>
<blockquote>
<ol start="2">
<li>Difference between codeql runner and codeql CLI?</li>
</ol>
</blockquote>
<p>The CodeQL runner is a wrapper around the CodeQL CLI. It’s designed to be used for simple integration into a CI/CD system, whereas the CodeQL CLI is designed for interactive analysis and query development purposes.</p>

**Comment by mkacymo** [comment]
<p>Hi,<br>
I have a similar issue: I want to use CodeQL and I’m working in a commercial setup that prevents me from uploading code to github. I tried contacting the github sales team, but was only offered solutions using github enterprise advanced security, although I clearly stated that I cannot upload any code to github and that I’m not interested in anything else but CodeQL licensing. I queried again and asked to make a license available, but my sales contact is unresponsive and refuses to escalate this matter internally or to connect me to the CodeQL team. To find a solution.<br>
Maybe someone here can help me:</p>
<ol>
<li>I want to use CodeQL to analyze code as part of security analyses/reviews</li>
<li>I do not own the code, but I am contracted to analyze and have the code owners permissions to do so.</li>
<li>I cannot upload this code to github or any other online storage.</li>
<li>I want to use CodeQL to define security queries. Thus I need a license allowing me to use the CLI for DB creation and then use CodeQL in VSCode for queries. Nothing more and nothing less, I only want to evaluate and use CodeQL technology.</li>
<li>This is not private but commercial use, thus I need a commercial license. So far the githib sales team refuses to offer one that does not enforce uploading and hosting via github enterprise. This is not possible with sensitive code and in my setup.</li>
</ol>
<p>Can anyone name a contact that is able to help? I really don’t want to violate license terms, so this lack of availability of a license option is currently a blocker.</p>

**Comment by RasmusWL** [comment]
<p>Hi <a class="mention" href="/mkacymo">@mkacymo</a>, thanks for your interest in CodeQL. I’m working as a CodeQL developer. While I don’t have the knowledge to answer your question directly, I’ll try to find the right people that can :blush:</p>
<p><em>PS. In the future, it would be better to open a new support topic instead of re-using a slightly related old one :wink:</em></p>

**Comment by niroshan** [comment]
<p>Hello <a class="mention" href="/mkacymo">@mkacymo</a> , I’d be happy to help you with your request.</p>
<p>my direct email is ########<em>at</em> github <em>dot</em> com , if you can drop me an email, I’ll talk you through how we can support your requirements.</p>
<p>Kind Regards,<br>
Niroshan</p>

**Comment by mkacymo** [comment]
<p>Hi <a class="mention" href="/rasmuswl">@rasmusWL</a>, thanks for connecting me to Niroshan. He’s now supporting me.<br>
Nevertheless there is still one big issue, where you could support:<br>
For some reason the sales representatives seem to consider CodeQL can only be used in combination with github hosting. Is there a misconception on my or their side? You could support here by clarifying if there is any technical obstacle in using CodeQL in a 100% offline setup (isolated lab), where the code never leaves the controlled local infrastructure. To my understanding there is nothing preventing this, but the lack of license availability for this use case in a commercial setup. Did I miss any technical obstacles?</p>

**Comment by RasmusWL** [comment]
<p>Hi <a class="mention" href="/mkacymo">@mkacymo</a>. On a technical level, CodeQL (CLI, VS Code extension, and such) can be used in a 100% offline setup.</p>

**Comment by bhuvi11** [comment]
<p>Hello Team,</p>
<p>Follow-up question regarding the usage of CodeQL.<br>
I am trying to use CodeQL commercially. Can that be done ? or should i get a license for it from GitHub to use so?</p>

**Comment by niroshan** [comment]
<p><a class="mention" href="/bhuvi11">@bhuvi11</a> , please feel free to drop me an email niroshan_at_github_dot_com</p>
<p>We can chat about your requirements and find something to help.</p>
